### DISCUSSION TITLE
Programatic way to close main window

### DISCUSSION THREAD

**Discussion by runrc** [cogentcore/core#1124] [Q&A] [answered]
What is the programmatic way to close the main window, possibly from another goroutine?

**Comment by runrc** [comment, marked as answer]
Use `system.TheApp.Quit()`

**Reply by kkoreilly** [reply]
Yes, to close all windows, you can call `system.TheApp.Quit` or `core.TheApp.Quit` (`core.TheApp` embeds `system.TheApp`). If you only want to close a specific window/scene, you can call `Scene.Close` (ex: `myWidget.Scene.Close()` to close the `Scene` that contains `myWidget`).

**Comment by runrc** [comment]
What is the recommended way to close the main window from another goroutine?
I have a signal handler that catches interrupt signals and needs to close the main window. 
However, I get a data race, when I call `system.TheApp.Quit()`

**Reply by kkoreilly** [reply]
You could try surrounding your `system.TheApp.Quit()` call with `AsyncLock` and `AsyncUnlock` calls on a widget in the main window; see https://www.cogentcore.org/core/basics/async.

**Reply by runrc** [reply]
Unfortunately, this causes a data race, I have raised a bug #1267
